webdriver-spec.html

```diff
@@ -1435,288 +1435,513 @@
+      each user agent will likely have conflicting methods of encoding the
+      user agent version and standardizing these schemes is beyond the scope of
+      this specification.
+
+      <dt>"<code>platformName</code>"
+      <dd><p>If <var>capability value</var> is not equal to
+      "<code>platformName</code>" entry in <var>matched capabilities</var>,
+      return null.
+
+      <dt>"<code>platformVersion</code>"
+      <dd><p>If <var>capability value</var> is not equal to
+      "<code>platformVersion</code>" entry in <var>matched capabilities</var>,
+      return null.
+
+      <dt>"<code>acceptSslCerts</code>"
+      <dd>
+        <p>If <var>capability value</var> is not a boolean,
+        return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+        <p>If <var>capability value</var> is <code>true</code> and the
+        <a>endpoint node</a> <em>does not</em> support insecure
+        SSL certificates, return null.
+
+        <p>Otherwise, set the "<code>acceptSslCerts</code>" entry in
+        <var>matched capabilities</var> to <var>capability value</var>.
+
+      <dt>"<code>pageLoadStrategy</code>"
+      <dd>
+        <p>If <var>capability value</var> is not a string,
+        return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+        <p>Otherwise, set the "<code>pageLoadStrategy</code>" entry in
+        <var>matched capabilities</var> to <var>capability value</var>.
+
+      <dt>"<code>proxy</code>"
+      <dd>
+        <p>If <var>capability value</var> is not a valid <a>Proxy Configuration</a>,
+        return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+        <p>If the <a>endpoint node</a> does not support the proxy configuration
+        defined in <var>capability value</var>, return null.
+
+        <p>Otherwise, set the "<code>proxy</code>" entry in
+        <var>matched capabilities</var> to <var>capability value</var>.
+
+      <dt>"<code>timeouts</code>"
+      <dd>
+        <p>If <var>capability value</var> is not a valid
+        <a>Session Timeouts Configuration</a> object,
+        return <a>error</a> with <a>error code</a> <a>invalid argument</a>.
+
+        <p>Otherwise, set the "<code>timeouts</code>" entry in
+        <var>matched capabilities</var> to <var>capability value</var>.
+
+      <dt><strong>Otherwise</strong>
+      <dd>
+        <p>If <var>capability name</var> is not recognized by the <a>endpoint node</a>,
+        return null.
+
+        <p>Let <var>processed value</var> be the result of implementation
+        specific steps to match on <var>capability name</var> with
+        <var>capability value</var>.
+
+        <p>If <var>processed value</var> is an error,
+        return <var>processed value</var>.
+
+        <p>Otherwise, create a new entry in <var>matched capabilities</var> with
+        name <var>capability name</var> and value <var>processed value</var>.
```

Comment:
I was trying to keep this two pass:

1.  Match and select session capabilities. That is what is described here.
2.  Initialize the session with matched capabilities. This is covered below.

The idea is matching and pre-checks are cheaper than trying to fully initialize a session. In other words, don't bother unpacking the profile in step (1) if there is another requested capability that the endpoint cannot satisfy.